backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/services/InterconnectionViewNodeToolSectionSwitch.java

```diff
@@ -64,18 +65,20 @@ public InterconnectionViewNodeToolSectionSwitch(IViewDiagramElementFinder cache)
 
     @Override
     public List<NodeToolSection> casePartUsage(PartUsage object) {
-        NodeDescription portNodeDescription = this.cache.getNodeDescription(this.descriptionNameGenerator.getBorderNodeName(SysmlPackage.eINSTANCE.getPortUsage())).get();
+        var optionalPortNodeDescription = this.cache.getNodeDescription(this.descriptionNameGenerator.getBorderNodeName(SysmlPackage.eINSTANCE.getPortUsage()));
 
         var createSection = this.toolDescriptionService.buildCreateSection(
                 this.toolDescriptionService.createNodeTool(this.getNodeDescription(SysmlPackage.eINSTANCE.getComment()),
                         SysmlPackage.eINSTANCE.getComment(), SysmlPackage.eINSTANCE.getOwningMembership(), null),
-                new ChildrenPartUsageCompartmentNodeToolProvider().create(null),
-                this.toolDescriptionService.createNodeTool(portNodeDescription, SysmlPackage.eINSTANCE.getPortUsage(), NodeContainmentKind.BORDER_NODE),
-                this.toolDescriptionService.createNodeToolWithDirection(portNodeDescription, SysmlPackage.eINSTANCE.getPortUsage(), NodeContainmentKind.BORDER_NODE, FeatureDirectionKind.IN),
-                this.toolDescriptionService.createNodeToolWithDirection(portNodeDescription, SysmlPackage.eINSTANCE.getPortUsage(), NodeContainmentKind.BORDER_NODE, FeatureDirectionKind.INOUT),
-                this.toolDescriptionService.createNodeToolWithDirection(portNodeDescription, SysmlPackage.eINSTANCE.getPortUsage(), NodeContainmentKind.BORDER_NODE, FeatureDirectionKind.OUT)
-
-        );
+                new ChildrenPartUsageCompartmentNodeToolProvider().create(null));
+        if (optionalPortNodeDescription.isPresent()) {
+            var portNodeDescription = optionalPortNodeDescription.get();
+            createSection.getNodeTools().add(this.toolDescriptionService.createNodeTool(portNodeDescription, SysmlPackage.eINSTANCE.getPortUsage(), NodeContainmentKind.BORDER_NODE));
+            createSection.getNodeTools().add(this.toolDescriptionService.createNodeToolWithDirection(portNodeDescription, SysmlPackage.eINSTANCE.getPortUsage(), NodeContainmentKind.BORDER_NODE, FeatureDirectionKind.IN));
+            createSection.getNodeTools().add(this.toolDescriptionService.createNodeToolWithDirection(portNodeDescription, SysmlPackage.eINSTANCE.getPortUsage(), NodeContainmentKind.BORDER_NODE, FeatureDirectionKind.INOUT));
+            createSection.getNodeTools().add(this.toolDescriptionService.createNodeToolWithDirection(portNodeDescription, SysmlPackage.eINSTANCE.getPortUsage(), NodeContainmentKind.BORDER_NODE, FeatureDirectionKind.OUT));
+        }
+        createSection.getNodeTools().add(new PartUsageBindingConnectorAsUsageNodeToolProvider((IVDescriptionNameGenerator) this.descriptionNameGenerator).create(this.cache));
```

Comment:
The unsafe cast here reveals there is something wrong with our approach. We shouldn't have to do this cast. Could you try to fix that please?

Reply:
I introduced a new interface to handle additional node name generation capabilities of the interconnection view description name generator.